File: functionaltests/source/resolve_java/src/main/java/Resolve.java
import mdk.Functions;
import mdk.MDK;
import mdk.Session;

public class Resolve {
    public static void main(String[] args) {
        MDK mdk = Functions.start();
        Session ssn = mdk.session();
        String address = ssn.resolve_until(args[0], "1.0.0", 10.0).address;
        System.out.println(address);
        mdk.stop();
    }
}

File: functionaltests/source/resolve.js
"use strict";

var MDK = require("datawire_mdk").mdk.start();

function showResult(result) {
    console.log(result.address);
    MDK.stop();
}

function showError(error) {
    console.log(error);
    MDK.stop();
}

function main() {
    setTimeout(function() {
        console.log("timeout");
        MDK.stop();
    }, 30000);
    var ssn = MDK.session();
    ssn.resolve_async(process.argv[2], "1.0.0").then(showResult, showError);
}

main();
